src/backend/InvenTree/InvenTree/cache.py

```diff
@@ -0,0 +1,105 @@
+"""Configuration options for InvenTree external cache."""
+
+import logging
+import socket
+
+import InvenTree.config
+import InvenTree.ready
+
+logger = logging.getLogger('inventree')
+
+
+def cache_setting(name, default=None, **kwargs):
+    """Return a cache setting."""
+    return InvenTree.config.get_setting(
+        f'INVENTREE_CACHE_{name.upper()}', f'cache.{name.lower()}', default, **kwargs
+    )
+
+
+def cache_host():
+    """Return the cache host address."""
+    return cache_setting('host', None)
+
+
+def cache_port():
+    """Return the cache port."""
+    return cache_setting('port', '6379', typecast=int)
+
+
+def is_global_cache_enabled():
+    """Check if the global cache is enabled.
+
+    - Test if the user has enabled and configured global cache
+    - Test if it is appropriate to enable global cache based on the current operation.
+    """
+    host = cache_host()
+
+    # Test if cache is enabled
+    # If the cache host is set, then the "default" action is to enable the cache
+    if not cache_setting('enabled', host is not None, typecast=bool):
+        return False
+
+    # Test if the cache is configured
+    if not cache_host():
+        logger.warning('Global cache is enabled, but no cache host is configured!')
+        return False
+
+    # The cache should not be used during certain operations
+    if any((
+        InvenTree.ready.isRunningBackup(),
+        InvenTree.ready.isRunningMigrations(),
+        InvenTree.ready.isRebuildingData(),
+        InvenTree.ready.isImportingData(),
+        InvenTree.ready.isInTestMode(),
```

Comment:
Q: How should caching be tested if it is always disabled in testing?

Reply:
We could override this like we do for other test cases with a special environment variable.